cql3/statements/cf_prop_defs.cc

```diff
@@ -188,6 +188,27 @@ std::optional<std::map<sstring, sstring>> cf_prop_defs::get_cdc_options() const
     return get_map(KW_CDC);
 }
 
+const schema::extensions_map& cf_prop_defs::get_schema_extensions(const db::extensions& exts) const {
+    if (_schema_extensions) {
+        return *_schema_extensions;
+    }
```

Comment:
I don't understand this - this function has a parameter, and yet the cached response doesn't depend on this parameter... If you call this function twice with different parameters, the second call will return wrong results!
Maybe you don't plan to call it twice with two different parameters (I only see one call in this patch, I don't know what will come next), but this is still a dangerous function to have around and who knows who will call it in future code. There isn't even a comment warning about not doing that (though I think in this case code changes will be more prudent than a comment).

Reply:
That's a good point, I see why it is seems dangerous to leave it like that. The extensions come from the the `database` object, which is shard-local. Those extensions shouldn't change during the execution.

I changed `get_schema_extensions` and `validate` methods to take `database` instead of `extensions` as an argument.

Reply:
Imho this is still not very kosher. Caching like this would be ok if database was a (non-changing) member, but here it is an arbitrary argument. 
If you want a cached `_schema_extensions` it should be generated as such at some point where the code flow guarantees that db is the same. 
And yes, I realize that who database is, is invariant in actual runtime, but remember all the talk of decoupling, multi-tenancy and moreover: testability. 
You technically only need the extensions once, so calculate them uncached, but keep the return value, and make `get_cdcoptions` take this extension object instead.

Reply:
I see. I changed it so that it constructs `schema::extensions_map` outside, and passes it to methods of `cf_prop_defs` as a parameter.